File: SupplierPortal.Web/Modules/Item/Offers/OffersColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using System.IO;

namespace SupplierPortal.Item.Columns;

[ColumnsScript("Item.Offers")]
[BasedOnRow(typeof(OffersRow), CheckNames = true)]
public class OffersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public short ItemCode { get; set; }
    [EditLink]
    public string ItemName { get; set; }
    public string CompanyName { get; set; }
    public string Email { get; set; }
    public Stream State { get; set; }
    public string ItemItemName { get; set; }
    public string SupplierCompanyName { get; set; }
}

File: SupplierPortal.Web/Modules/ServerTypes/Item/OffersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OffersRow } from "./OffersRow";

export interface OffersColumns {
    Id: Column<OffersRow>;
    ItemCode: Column<OffersRow>;
    ItemName: Column<OffersRow>;
    CompanyName: Column<OffersRow>;
    Email: Column<OffersRow>;
    State: Column<OffersRow>;
    ItemItemName: Column<OffersRow>;
    SupplierCompanyName: Column<OffersRow>;
}

export class OffersColumns extends ColumnsBase<OffersRow> {
    static readonly columnsKey = 'Item.Offers';
    static readonly Fields = fieldsProxy<OffersColumns>();
}